sycl/source/detail/graph_impl.hpp

```diff
@@ -88,6 +88,26 @@ class node_impl {
             std::unique_ptr<sycl::detail::CG> &&CommandGroup)
       : MCGType(CGType), MCommandGroup(std::move(CommandGroup)) {}
 
+  /// Tests if two nodes have the same content,
+  /// i.e. same command group
+  /// @param Node node to compare with
```

Comment:
Do we need both this and `isSimilar(std::shared_ptr<node_impl> Node)`? Functions look very similar, and if we do need both can we share code somehow so that one calls the other?

Reply:
Yes we need both because operator== check only the content of those nodes while isSimilar checks the content but also that the nodes have the same number of successors and predecessors.
The first allows comparing nodes regardless of their environment (useful to compare nodes after graph changes such as adding a subgraph).
The second allows comparing two graphs node by node. 
However, I made isSimilar calls operator== to avoid duplicate code.

Reply:
Thanks for the explanation, using  `operator==` in `isSimilar` looks like a good way of doing it 👍